test/health_failures_test_case.py

```diff
@@ -0,0 +1,117 @@
+from contextlib import contextmanager
+import os
+import json
+from unittest import mock
+from uuid import uuid4
+
+import boto3
+from mock import (
+    MagicMock,
+    patch,
+)
+from more_itertools import chunked
+from moto import (
+    mock_sts,
+    mock_sqs,
+    mock_dynamodb2,
+)
+import requests
+
+from app_test_case import LocalAppTestCase
+from azul import config
+from azul.health import HealthController
+from azul.json_freeze import (
+    freeze,
+    sort_frozen,
+)
+from azul.modules import load_app_module
+from retorts import ResponsesHelper
+
+
+class TestHealthFailures(LocalAppTestCase):
+
+    @classmethod
+    def lambda_name(cls) -> str:
+        return 'service'
+
+    @contextmanager
+    def _make_database(self):
+        database = boto3.resource('dynamodb', region_name='us-east-1')
+        table_settings = {
+            'TableName': config.dynamo_failure_message_table_name,
+            'KeySchema': [
+                {
+                    'AttributeName': 'MessageType',
+                    'KeyType': 'HASH'
+                },
+                {
+                    'AttributeName': 'SentTimeMessageId',
+                    'KeyType': 'RANGE'
+                }
+            ],
+            'AttributeDefinitions': [
+                {
+                    'AttributeName': 'MessageType',
+                    'AttributeType': 'S'
+                },
+                {
+                    'AttributeName': 'SentTimeMessageId',
+                    'AttributeType': 'S'
+                }
+            ],
+            'ProvisionedThroughput': {
+                'ReadCapacityUnits': 5,
+                'WriteCapacityUnits': 5
+            }
+        }
+        database.create_table(**table_settings)
+        with mock.patch.dict(os.environ, AWS_DEFAULT_REGION='us-east-1'):
+            try:
+                yield
+            finally:
+                table = database.Table(config.dynamo_failure_message_table_name)
+                table.delete()
+
+    @mock_sts
+    @mock_sqs
+    @mock_dynamodb2
+    def test_failures_endpoint(self):
+        indexer_app = load_app_module('indexer')
+        sqs = boto3.resource('sqs', region_name='us-east-1')
+        sqs.create_queue(QueueName=config.fail_queue_name)
+        fail_queue = sqs.get_queue_by_name(QueueName=config.fail_queue_name)
+        with patch('azul.time.RemainingLambdaContextTime.get', return_value=1000):
```

Comment:
How long do you expect the body of this context manager to run for in the worst case?

Reply:
I forgot to change `return_value` to 5 seconds to per subtest case.

Reply:
Can you answer my question, please?

Reply:
5 seconds to per subtest case.

Reply:
That can't be right. You are patching `receive_message_wait_time` to zero. Please describe how you arrived at that 5s estimate.

Reply:
While `receive_message_wait_time` to zero, the processes after waiting for the messages to be received would a few seconds more than 0. I thought 5 seconds would be the longest worst case while still being strict margin.

Reply:
I don't understand what you mean by "while still being strict margin". Can you please elaborate.

Can you explain what would take 5s when all the test is doing is using in-memory mocks of DynamodDB and SQS. 

I am also concerned that you are retroactively pretending have thought this fully through. I think you did not but I have no proof. It is just that your estimate of 5s is suspiciously identical to the 5s default for `receive_message_wait_time` and also astronomically high considering the use of mocks.

Reply:
Yeah, I based my answer from the value given in `receive_message_wait_time`.